src/include/daos_obj.h

```diff
@@ -1129,13 +1129,97 @@ daos_oit_close(daos_handle_t oh, daos_event_t *ev);
  * \return		These values will be returned by \a ev::ev_error in
  *			non-blocking mode:
  *			0		Success
- *			-DER_NO_HDL	Invalid object open handle
+ *			-DER_NO_HDL	Invalid OIT open handle
  *			-DER_INVAL	Invalid parameter
  */
 int
 daos_oit_list(daos_handle_t oh, daos_obj_id_t *oids, uint32_t *oids_nr,
 	      daos_anchor_t *anchor, daos_event_t *ev);
 
+#define DAOS_OIT_MARKER_MAX_LEN	(32)
+
+/**
+ * Mark an object ID in the Object Index Table (OIT).
+ *
+ * \param[in]	oh	OIT open handle.
+ * \param[in]	oid	object ID in the OIT.
+ * \param[in]	marker	the data/status to be marked for the OID, the max valid length (in bytes)
+ *			is DAOS_OIT_MARKER_MAX_LEN. NULL is invalid parameter.
+ * \param[in]	ev	Completion event, it is optional and can be NULL.
+ *			Function will run in blocking mode if \a ev is NULL.
+ *
+ * \return		These values will be returned by \a ev::ev_error in
+ *			non-blocking mode:
+ *			0		Success
+ *			-DER_NO_HDL	Invalid OIT open handle
+ *			-DER_INVAL	Invalid parameter
+ *			-DER_NONEXIST	Input OID not in the OIT.
+ */
+int
+daos_oit_mark(daos_handle_t oh, daos_obj_id_t oid, d_iov_t *marker, daos_event_t *ev);
+
+/**
+ * Enumerate unmarked object IDs snapshotted by the Object Index Table (OIT)
+ *
+ * \param[in]	oh	OIT open handle.
+ * \param[out]	oids	Returned OIDs
+ * \param[out]	oids_nr	Number of returned OIDs
+ * \param[in,out]
+ *		anchor	Hash anchor for the next call, it should be set to
+ *			zeroes for the first call, it should not be changed
+ *			by caller between calls.
+ * \param[in]	ev	Completion event, it is optional and can be NULL.
+ *			Function will run in blocking mode if \a ev is NULL.
+ *
+ * \return		These values will be returned by \a ev::ev_error in
+ *			non-blocking mode:
+ *			0		Success
+ *			-DER_NO_HDL	Invalid OIT open handle
+ *			-DER_INVAL	Invalid parameter
+ */
+int
+daos_oit_list_unmarked(daos_handle_t oh, daos_obj_id_t *oids, uint32_t *oids_nr,
+		       daos_anchor_t *anchor, daos_event_t *ev);
```

Comment:
Just curious, why oit_mark for single OID, and list_unmarked for multiple OIDs? I mean why not both for multiple OIDs or single OID?

Reply:
Ah, this is list, instead of unmark()
But this seems duplicate with daos_oit_list_filter(). I mean list_unmarked = list_filter(unmark_filter_cb()).

Reply:
so to be clear, this is not unmarking, but just returning the oids that are unmarked.
it's duplicate in functionality to calling list_filter with the cb checking the buffer if it exists or not, but as i said, in performance, it's better since you can just return the unmarked oids from the server side.

that being said, @liuxuezhao we probably need an additional API to clear the mark?

Reply:
I mean you can ship "the filter" to the server by "flag or sth" anyway? it does not matter if the filter is for unmark oit or filtering sth else.

Reply:
how can you ship a user defined function CB?

Reply:
filter_cb() will somehow define the filter by marker?  then  unmarked case can send to the server by flag. Anyway, I thought list_filter() needs to ship "the filter" to server for better performance in some cases? no?

Reply:
sorry not sure what you mean anymore.. 
only the user knows the marker format and defines the callback, so how can it be shipped?

Reply:
I mean it can define some flags, like filter_unmark_oid, and export to the api user, then the user can use these flag to indicate the specific filtering, like unmark or its own defined filtering.
 
I mean list_filter() needs to "ship the filter" to the server in some cases, then why not we unify the interface here?

Reply:
but you cannot define all possible things a user would want to do in a cb in some flags.
the data is user defined and we don't know what's in it to be able to define flags that can be applied other than: does the mark exist or no.

Reply:
Oh, only specific flags, which you want to ship to the server. like unmark etc.  For other filter, it will do totally client side filtering.

Reply:
is there any other flag other than unmarked you can think of :-) ?
i agree if there are other flags that can be considered useful, we could change that.

Reply:
I am not sure now. but I looks like  "marker" seems as an EA(extend attribute) for the object, so I assume there will be other filter to analyze the EA?

Reply:
@mchaarawi "we probably need an additional API to clear the mark?" 
we probably can reuse daos_oit_mark() API, so when the input "d_iov_t *marker" is NULL or the "marker->iov_buf" is NULL means clear the mark?

@wangdi1 I am not sure if we are going to provide "EA" similar concept for object, current API only to support dfs usage for middleware consistency.